### PR TITLE
Update design of cart and checkout sidebars

### LINKED ISSUE #4093: Changes to the Cart + Checkout block sidebar
Hi folks, this request is to improve some of the UX in our Cart + Checkout block experience for buyers.
 
Sections that will be changed: 

- Shipping title
- Taxes
- Coupon input

**Shipping**
Group shipping information into its own section and remove title style. 

**Taxes**
Isolate taxes into its own section and position underneath the shipping options section. 

**Coupon input** 
Shift position of this Coupon input upwards to appear under the subtotals section.

These changes should be reflected in _both_ the Cart and Checkout views. I have attached visual comparisons below. 

cart-comparison

checkout-comparison

Thank you and ping me if you have any questions :) 

Figma design files: eaQcoMs7E8Ys0YgxfkOKxR-fi-727%3A1243

### PR DESCRIPTION
**Pull request by opr**
### Summary of changes
@mikejolley Made these changes:
- Reorder the output of components in `/full-cart/index.tsx` so the Coupon input is in the correct place
- Add styles to reflect new design changes
- Update wording in Coupon input and Shipping methods components
- Add order summary heading to Checkout block sidebar

Thanks Mike!

I made these:
- Modify `src/BlockTypes/Cart.php` to ensure the `displayItemizedTaxes` option is available to the Cart block.
- Modify `CartSchema.php` to include a `rate` option on the tax line. This is needed to display the rate alongside tax names e.g <pre>CA Sales tax **7.5%**</pre> as per the design. This is enabled on a per-block basis.
- Modify `TotalsTaxes` component to add the itemised tax rates as children of the "main" tax line. There is also a stylesheet to ensure these display as per the design.
- Add an attribute to the Cart and Checkout blocks: `showRateAfterTaxName` this can be used to toggle whether the rate is shown after a tax's name.
- Updated the preview data of the cart to include some taxes.
- Added a `ShippingVia` component to show the rates that will be used to send the packages.
- Updated the Cart and Checkout blocks `edit.js` file to ensure the correct tax rates are displayed based on store settings.
- Added JS tests to the Cart and Checkout blocks to ensure these new options are working as intended. These tests use snapshots that I inspected myself 👀 

Fixes #4093

#### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [x] I've tested using a screen reader
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Cart sidebar
| Before | After |
|---|---|
| image [Full size image](https://user-images.githubusercontent.com/5656702/117444911-fabed200-af31-11eb-84a4-e1fffdded141.png) | image [Full size image](https://user-images.githubusercontent.com/5656702/118136128-addc6f00-b3fb-11eb-9a52-43133f823648.png)

#### Checkout sidebar
| Before | After |
|---|---|
| image [Full size image](https://user-images.githubusercontent.com/5656702/117698354-8f883080-b1bb-11eb-9787-f78ae2e63481.png) |  image [Full size image](https://user-images.githubusercontent.com/5656702/118120982-afe90280-b3e8-11eb-8ca4-4526db34b79b.png) |

#### Setup steps
1. Ensure some items in your store are taxable, and add a few of them to your cart, for bonus points you can set up additional tax rates and add products from each rate to your cart.
2. Ensure you have a coupon available to use in your store.
3. Go to `WooCommerce -> Settings -> Tax` and set `Display tax totals` to `Itemized` and `Display prices during basket and checkout` to `Excluding Tax`.
2. Using the designs presented in #4093 - Figma link: eaQcoMs7E8Ys0YgxfkOKxR-fi-727%3A1243 do the following:

#### Editor testing - do these in the block editor

**These instructions are for both the Cart and Checkout block!**
1. Add the block and ensure it matches the designs from the Figma file.
2. Ensure you have the option to `Show rate after tax name` in the editor sidebar.
2. Toggle this on and off and verify the rate percentage is shown in the Taxes section of the block preview.
4. Go to `WooCommerce -> Settings -> Tax` and set `Display tax totals` to `As a single total` and then ensure the option to `Show rate after tax name` is no longer available in the Cart and Checkout blocks.
5. Ensure the individual tax lines are not shown in the block preview.
6. Set it back to `Itemized`.
6.  Go to `WooCommerce -> Settings -> Tax` and set `Display prices during basket and checkout` to `Including Tax`.
7. Ensure the Taxes section is not shown, but that the amount of tax is shown under the total.
8. Set it back to `Excluding Tax`.
8. While doing the next set of testing instructions, experiment with combinations of these settings ensuring they work correctly.

#### Front-end testing
1. Go to the Cart/Checkout and ensure the rendered cart matches the Figma file. Ensure you have the correct Taxes section based on your configuration.
4. Go to `WooCommerce -> Settings -> Tax` and set `Display tax totals` to `As a single total` and then ensure the Taxes section of the sidebar does not include a further breakdown of rates.
5. Set it back to `Itemized` and ensure the rates are broken down correctly.
6. Toggle the `Show rate after tax name` option in the block and ensure the changes are reflected on the front-end.
6.  Go to `WooCommerce -> Settings -> Tax` and set `Display prices during basket and checkout` to `Including Tax`.
7. Ensure the Taxes section is not shown, but that the amount of tax is shown under the total.
8. Select different shipping methods and ensure the `via X` updates in the Shipping section of the sidebar. Do this in both Cart and Checkout.
9. Add a coupon and ensure the discounts section shows the reduction amount.

#### Automated testing
1. Ensure JS tests are passing.

### Changelog

> Update the display of the sidebar/order summary in the Cart and Checkout blocks.

### CODE REVIEW

**Comment by mikejolley** [comment]
I think we still need feedback on the issue regarding the "order summary" heading right? In the rework we have it duplicated. It toggles visibility of the items. If we remove it, it impacts handheld views which may or may not be problematic.

**Comment by opr** [comment]
> I think we still need feedback on the issue regarding the "order summary" heading right? In the rework we have it duplicated. It toggles visibility of the items. If we remove it, it impacts handheld views which may or may not be problematic.

Oh yep you're right. I'll set it back to draft while waiting for feedback from @vivialice . For reference Mike's original comment/question is here: https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4093#issuecomment-833489638

**Comment by vivialice** [comment]
👋  Thanks for flagging this - can we remove the duplicate and keep the title as-is (with the dropdown below the divider)

**Comment by opr** [comment]
> 👋 Thanks for flagging this - can we remove the duplicate and keep the title as-is (with the dropdown below the divider)

Thanks @vivialice ! I implemented that, did you mean like this? It looks a bit strange when collapsed, what do you think?

| Open | Collapsed |
|---|---|
|  |   |

**Comment by vivialice** [comment]
Haha sorry, I wasn't clear. I meant to change it back to the original 😄

File: assets/js/base/components/cart-checkout/totals/shipping/shipping-via.tsx
Comment (mikejolley) [inline assets/js/base/components/cart-checkout/totals/shipping/shipping-via.tsx]: eslint warning looks valid-should return react element.

File: assets/js/base/components/cart-checkout/totals/shipping/shipping-via.tsx
Comment (mikejolley) [inline assets/js/base/components/cart-checkout/totals/shipping/shipping-via.tsx:12]: Is a dedicated classname needed for these new descriptions?

File: assets/js/base/context/hooks/cart/use-store-cart-coupons.ts
Comment (mikejolley) [inline assets/js/base/context/hooks/cart/use-store-cart-coupons.ts]: Added but unused.

File: assets/js/blocks/cart-checkout/cart/attributes.js
Comment (mikejolley) [inline assets/js/blocks/cart-checkout/cart/attributes.js]: Should we default this to true? Why is it based on `displayCartPricesIncludingTax`?

File: packages/checkout/totals/taxes/index.tsx
Comment (mikejolley) [inline packages/checkout/totals/taxes/index.tsx]: Another eslint warning.

**Review by mikejolley — APPROVED** [review]
Tests well 🥇 

I noticed a few remaining eslint warnings that could be fixed before merging, and a few small comments, but I am preapproving.

File: assets/js/blocks/cart-checkout/cart/attributes.js
Comment (opr) [reply assets/js/blocks/cart-checkout/cart/attributes.js]: Good spot! This was a mistake. I think we can just default straight to a boolean value as there's nothing on the back end, besides the inspector control that can decide this. Do you think it should default to `true`? I think defaulting to `false` would be better, that way it's an opt-in feature rather than opt-out.

Comment (mikejolley) [reply assets/js/blocks/cart-checkout/cart/attributes.js]: Thought as much!

I think true, because its something that can be disabled should someone have customised the names, but I think it's more likely that the % is beneficial. I think we should default it to on so the default experience matches the designs.

Sometimes itemised taxes look weird without this, e.g. on my test store I had "Taxes" with a sub-heading below of "Tax". With this true, it was "Tax 20%" which is more useful :)

File: assets/js/base/components/cart-checkout/totals/shipping/shipping-via.tsx
Comment (opr) [reply assets/js/base/components/cart-checkout/totals/shipping/shipping-via.tsx:12]: There is a way to target this element without adding the dedicated class, but the CSS rule looks like this.

`.wc-block-components-totals-shipping .wc-block-components-totals-item__description > .wc-block-components-totals-item__description`

I'm not such a fan of it and would prefer to keep the dedicated class name.

**Pull request merged**
